### ISSUE TITLE
Mailbox does not implement `TryInto<Mailbox>`

### ISSUE THREAD

**Issue by felinira** [pimalaya/imap-client#12]
Many `Client` methods expect a `impl TryInto<Mailbox<'_>, Error = ValidationError>`, but Mailbox does not actually implement this trait (only infallibly through the blanket `From<T> for T` implementation in Rust std). Not sure whether to report it at imap-types or here, but this makes it a little difficult to pass an already existing mailbox type :) And I don't think you can implement `TryFrom` twice, as Error is an associated type, so either way it probably needs to be solved here. Maybe through some new trait that allows both `Into` and `TryInto`, or by delegating the responsibility of conversion to the caller.

**Comment by soywod** [comment]
Yes indeed, we will not be able to implement `TryFrom` for itself. I often face this dilemna and I'm not sure which solution is the best between converting inside or outside. Both have pros and cons. Any advice?

**Comment by duesee** [comment]
Hm... yeah... I decided against handling mUTF-7 on the imap-types level because I was afraid to confine ourselfes too much.

A `Mailbox` is (simplified ) an `AString` which means an `Atom` or a `String` (which means a `Quoted` or a `Literal`). What this says to me is: A Mailbox is a bunch of bytes (w/o 0x00).

imap-types will try all variants from atom, quoted, literal, error (due to 0x00) and send the mailbox as such.

@felinira, can you open an issue in the imap-coded repo and provide more details? We stumbled upon mUTF-8 already but so far I was afraid to tackle it and just implemented the RFC in it's most generic form. But: This should never had lead to a server error. This is what I want to promise with imap-types :-)

**Comment by duesee** [comment]
By the way: If you feel something could be off with a client/server, feel free to discuss via an issue in https://github.com/modern-email/defects.
